shotover-proxy/src/tls.rs

```diff
@@ -101,7 +101,10 @@ impl TlsConnector {
         let ssl = self.connector.configure()?.into_ssl("localhost")?;
 
         let mut ssl_stream = SslStream::new(ssl, tcp_stream)?;
-        Pin::new(&mut ssl_stream).connect().await?;
+        Pin::new(&mut ssl_stream)
+            .connect()
+            .await
+            .map_err(|e| anyhow!(e).context("Failed to connect to TLS connection"))?;
```

Comment:
Nitpick: Connecting to a connection... sounds odd. How about "Failed to establish TLS connection"?

Reply:
Thats a good point.
Another thing to keep in mind is the direction is important here.
We failed to connect to something rather then failing to accept a connection.

I think I'll go with "Failed to establish TLS connection to destination"